packages/langium/src/lsp/language-server.ts

```diff
@@ -326,6 +333,64 @@ export function addSignatureHelpHandler(connection: Connection, services: Langiu
     ));
 }
 
+function addCallHierarchyHandler(connection: Connection, services: LangiumSharedServices): void {
+    const errorMessage = 'No call hierarchy provider registered';
+    connection.languages.callHierarchy.onPrepare(createServerRequestHandler(
+        (services, document, params, cancelToken) => {
+            if (services.lsp.CallHierarchyProvider) {
+                return services.lsp.CallHierarchyProvider.prepareCallHierarchy(document, params, cancelToken);
```

Comment:
Suggestion: Instead of returning `null` (which we don't do anywhere else in the framework), we should instead return `undefined` from our service and do the following here. The same applies to the other service methods as well.
```suggestion
                return services.lsp.CallHierarchyProvider.prepareCallHierarchy(document, params, cancelToken) ?? null;
```